app/lib/foreman_deployments/tasks/search_task_definition.rb

```diff
@@ -7,6 +7,26 @@ def run
           results = SerializableArray.new(results)
           SearchTaskDefinition.create_output(results, output)
         end
+
+        def self.dereference_output(storage, path)
```

Comment:
I like the idea of having an object that takes care of (de)serialization of actions' inputs/outputs. There's an issue open for that in deployments http://projects.theforeman.org/issues/12172 but it would certainly be cleaner to have it supported directly in dynflow.

Regarding @ShimShtein's thoughts in the latest comment it might be interesting to introduce the concept of transformations. Since `OutputReference` takes care of dereferencing itself (correct me if I'm wrong) I think we could add support for using transform functions and have the both concepts could live side by side. Action writers can then choose what's more appropriate for them.

The latest "thread-pool-like" idea seems quite against the dynflow's design to me. As @iNecas mentions, planning heeds to happen before the execution.
BTW specifically in the example with the host I think the `DeleteHost` action should rather be part of the planned rollback mechanism (https://github.com/Dynflow/dynflow/issues/87).